pkg/mcclient/mcclient.go

```diff
@@ -0,0 +1,158 @@
+package mcclient
+
+// kubestellar cluster-aware client impl
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	log "k8s.io/klog/v2"
+
+	"github.com/kcp-dev/edge-mc/pkg/apis/edge/v1alpha1"
+	ksclientset "github.com/kcp-dev/edge-mc/pkg/client/clientset/versioned"
+	ksinformers "github.com/kcp-dev/edge-mc/pkg/client/informers/externalversions"
+	mcclient "github.com/kcp-dev/edge-mc/pkg/mcclient/clientset"
+	"github.com/kcp-dev/edge-mc/pkg/mcclient/listwatch"
+)
+
+type KubestellarClusterInterface interface {
+	// Cluster returns clientset for given cluster.
+	Cluster(name string) (client mcclient.Interface)
+	// ConfigForCluster returns rest config for given cluster.
+	ConfigForCluster(name string) (*rest.Config, error)
+	// CrossClusterListWatch returns cross-cluster ListWatch
+	CrossClusterListWatch(gv schema.GroupVersion, resource string, namespace string, fieldSelector fields.Selector) *listwatch.CrossClusterListerWatcher
+}
+
+type multiClusterClient struct {
+	configs    map[string]*rest.Config
+	clientsets map[string]*mcclient.Clientset
+	lock       sync.Mutex
+}
+
+func (mcc *multiClusterClient) Cluster(name string) mcclient.Interface {
+	mcc.lock.Lock()
+	defer mcc.lock.Unlock()
+	clientset, ok := mcc.clientsets[name]
+	if !ok {
+		//TODO change to ClusterOrDie and panic? return an error?
```

Comment:
Return an error here.  Handling should be the client's decision.

Reply:
If we return an error here then something like `mcclient.Cluster("cluster1").KS().EdgeV1alpha1()...` will not compile and the client will have to first check the error from `mcclient.Cluster`
Also the current clustered clientset code [panics on error](https://github.com/kcp-dev/edge-mc/blob/main/pkg/client/clientset/versioned/cluster/clientset.go#L74)